hyperspy/signal.py

```diff
@@ -3111,6 +3112,83 @@ def crop(self, axis, start=None, end=None, convert_units=False):
         if convert_units:
             self.axes_manager.convert_units(axis)
 
+    def interpolate_on_axis(self,
+                            new_axis,
+                            replace_axis_index,
+                            inplace=False,
+                            set_navigate=True,
+                            **kwargs):
+        """Replaces the axis specified by replace_axis_idx with the provided new_axis
+        and interpolates data accordingly.
+
+        Parameters
+        ----------
+        new_axis : UniformDataAxis, DataAxis or FunctionalDataAxis
+            Axis which replaces the axis specified with replace_index.
+            If this new axis exceeds the range of the old axis,
+            a warning is raised that the data will be extrapolated.
+
+        replace_axis_index : int
+            Specifies the axis which will be replaced using the index of the
+            axis in `axes_manager`.
+
+        inplace : bool, default=False
+            If ``True`` the data of `self` is replaced by the result and
+            the axis is changed inplace. Otherwise `self` is not changed
+            and a new signal with the changes incorporated is returned.
+
+        set_navigate : bool, default=True
+            If ``True``, sets the `navigate` attribute of `new_axis` to the value given
+            by the old axis.
+
+        **kwargs: dict
+            Extra keywords passed to :py:func:`scipy.interpolate.interp1d`.
+            Extracted are: `kind` (default: linear), `bounds_error` (default: False)
+            and `fill_value` (default: extrapolate).
+
+        Returns
+        -------
+        s : :py:class:`~hyperspy.signal.BaseSignal` (or subclass)
+            A copy of the object with the axis exchanged and the data interpolated.
+            This only occurs when inplace is set to ``False``, otherwise nothing is returned.
+        """
+        kind = kwargs.get("kind", "linear")
+        bounds_error = kwargs.get("bounds_error", False)
+        fill_value = kwargs.get("fill_value", "extrapolate")
+
+        old_axis = self.axes_manager[replace_axis_index]
+        if old_axis.low_value > new_axis.low_value or old_axis.high_value < new_axis.high_value:
+            _logger.warning(
+                "The specified new axis exceeds the range of the to be replaced old axis. "
+                "The data will be extrapolated if not specified otherwise via fill_value/bounds_error"
+            )
+
+        nav_dim = self.axes_manager.navigation_dimension
+        sig_dim = self.axes_manager.signal_dimension
+        if old_axis.navigate:
+            set_axis_idx = (nav_dim - 1) - replace_axis_index
+        else:
+            set_axis_idx = (2*nav_dim + sig_dim - 1) - replace_axis_index
+
+        if set_navigate:
+            new_axis.navigate = old_axis.navigate
+
+        interpolator = interp1d(
```

Comment:
`interp1d` is a legacy function in SciPy that will be deprecated in the future: https://docs.scipy.org/doc/scipy/tutorial/interpolate/1D.html#legacy-interface-for-1-d-interpolation-interp1d

We should either use `[numpy.interp](https://numpy.org/devdocs/reference/generated/numpy.interp.html#numpy.interp)` (corresponds to current default behaviour) or one of the more specific scipy interpolation functions: https://docs.scipy.org/doc/scipy/tutorial/interpolate/1D.html

If we want to make the type of interpolation a parameter, we might have to handle that in our function (linear, spline and pchip seem interesting on first sight)?

Reply:
Maybe better to look at this in a separate PR?

Reply:
Of course, I'll open an issue. Any comments on this PR?